staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go

```diff
@@ -421,6 +422,33 @@ func (s *store) Count(key string) (int64, error) {
 	return getResp.Count, nil
 }
 
+// LastResourceVersion finds the last modified resource version for a given key range and returns it
+// or an error.
+func (s *store) LastResourceVersion(ctx context.Context, keyPrefix string, from string) (string, error) {
+	fromRV, err := s.versioner.ParseListResourceVersion(from)
+	if err != nil {
+		return "", err
+	}
+	key := path.Join(s.pathPrefix, keyPrefix)
+	getResp, err := s.client.KV.Get(
```

Comment:
how does this handle deleted resources? for example:

add /pod/a (rv=1)
add /pod/b (rv=2)
add /pod/c (rv=3)
update /pod/a (rv=4)
update /pod/b (rv=5)
delete /pod/c (rv=6)

say the watch cache has observed up to rv=3

will LastResourceVersion only see /pod/a and /pod/b, and determine that rv=5 is the lastRev?

Reply:
You're right, it will.  Back to the drawing board.  I know the store has tombstones but we may need an API change to get them out.